source/elements/oneMKL/source/domains/spblas/setcsrstructure.rst

```diff
@@ -33,8 +33,9 @@ set_csr_data (Buffer version)
 
       void set_csr_data (sycl::queue                           &queue,
                          oneapi::mkl::sparse::matrix_handle_t  handle,
-                         intType                               num_rows,
-                         intType                               num_cols,
+                         intType                               nrows,
+                         intType                               ncols,
+                         intType                               nnz,
```

Comment:
```suggestion
                         const intType                         nrows,
                         const intType                         ncols,
                         const intType                         nnz,
```

Reply:
It seems like we should have const on these.  Is there an argument for not doing it ?

Reply:
Intel oneMKL has `const`, but rocSPARSE, or cuSPARSE do not have a `const` in their CSR matrix creation APIs. See:
* [oneapi::mkl::sparse::set_csr_data()](https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-dpcpp/2023-2/oneapi-mkl-sparse-set-csr-data.html)
* [rocsparse_create_csr_descr()](https://rocm.docs.amd.com/projects/rocSPARSE/en/latest/auxiliary.html#rocsparse-create-csr-descr)
* [cusparseCreateCsr()](https://docs.nvidia.com/cuda/cusparse/index.html#cusparsecreatecsr)

Coming from C++, I think we should have a `const` in the spec, and so should, I believe, cuSPARSE and rocSPARSE. Because these are just integers, I don't think it matters either for functional correctness or performance reasons whether or not we have a `const` in there.

cuSPARSE/rocSPARSE backends only have `int64_t` type on nrows/ncols/nnz and not both {`int32_t`, `int64_t`} options, so libraries implementing this oneMKL Spec must anyway create a copy from `int32_t` to `int64_t` when dispatching to those backends, We might as well go ahead and add the `const` as you suggested here.

Reply:
Actually on second thought, we also have the enum, `oneapi::mkl::index_base` in this API and additional enums in many other APIs which are not `const` (yet). We should be more disciplined and uniform about how and where we add the `const` keyword. I think in the next version of the spec, we should fix the `const`-ness of variables in all the APIs instead of just some. If only we had more time, we could have done it right in the upcoming v1.3. I'd like to keep it non-const for now and deal with this later.